lib/filesystem.nix

```
@@ -306,52 +306,65 @@ in
         As a result, directories with no `.nix` files (including empty
         directories) will be transformed into empty attribute sets.
 
-    # Inputs
-
-    Structured function argument
-
-    : Attribute set containing the following attributes.
-      Additional attributes are ignored.
-
-      `callPackage`
-
-      : `pkgs.callPackage`
-
-        Type: `Path -> AttrSet -> a`
-
-      `directory`
+    # Type
 
-      : The directory to read package files from
+    ```
+    packagesFromDirectoryRecursive :: {
+      callPackage :: Path -> {} -> a,
+      directory :: Path,
+      ...
+    } -> AttrSet
+    ```
 
-        Type: `Path`
+    # Inputs
 
+    `callPackage`
+    : The function used to convert a nix file's path, into a leaf of the attrset.
+      It is typically the `callPackage` function, taken from either `pkgs` or a new scope corresponding to the `directory`.
 
-    # Type
+    `directory`
+    : The directory to read package files from.
 
-    ```
-    packagesFromDirectoryRecursive :: AttrSet -> AttrSet
-    ```
 
     # Examples
     :::{.example}
-    ## `lib.filesystem.packagesFromDirectoryRecursive` usage example
+    ## Basic use of `lib.packagesFromDirectoryRecursive`
 
     ```nix
     packagesFromDirectoryRecursive {
       inherit (pkgs) callPackage;
       directory = ./my-packages;
     }
     => { ... }
+    ```
+
+    In this case, `callPackage` will only search `pkgs` for a file's input parameters.
+    In other words, a file cannot refer to another file in the directory in its input parameters.
+    :::
 
+    :::{.example}
+    ## Create a scope for the nix files found in a directory
+    ```nix
     lib.makeScope pkgs.newScope (
       self: packagesFromDirectoryRecursive {
-        callPackage = self.callPackage;
+        inherit (self) callPackage;
         directory = ./my-packages;
       }
     )
     => { ... }
     ```
 
+    Here, if `directory` contains `awoo.nix` and `owo/uwu.nix`, with the latter expecting `{ awoo }` as a parameter,
+    `callPackage` will resolve it as expected.
+
+    :::{.note}
+    `awoo.nix` can get `owo` (as an attrset) in input, but not `uwu` directly.
+    :::
+    :::{.warning}
+    As of now, `lib.packagesFromDirectoryRecursive` cannot create nested scopes for sub-directories.
+
+    In this example, a file `owo/bonk.nix` cannot expect `uwu` (from the same sub-directory) as a direct parameter.
+    :::
```

```suggestion
    ::::{.note}
    `awoo.nix` can get `owo` (as an attrset) in input, but not `uwu` directly.
    ::::
    
    ::::{.warning}
    As of now, `lib.packagesFromDirectoryRecursive` cannot create nested scopes for sub-directories.

    In this example, a file `owo/bonk.nix` cannot expect `uwu` (from the same sub-directory) as a direct parameter.
    ::::
```

Not sure this will nest properly with the surrounding `:::{.example}` otherwise. Did you try building it with `devmode` from `nix-shell ./doc`?

---

Its not nested, is it?